build/signDSCModules_copy.sh

```diff
@@ -0,0 +1,198 @@
+#!/bin/bash
+
+# Assume we're running from superproject (with version file)
+
+if [ ! -f omsagent.version ]; then
+    echo "Is current directory wrong? Can find version file omsagent.version ..."
+    exit 1
+fi
+
+# Read in the version file
+
+source omsagent.version
+
+if [ -z "$OMS_BUILDVERSION_MAJOR" -o -z "$OMS_BUILDVERSION_MINOR" -o -z "$OMS_BUILDVERSION_PATCH" -o -z "$OMS_BUILDVERSION_BUILDNR" ]; then
+    echo "Missing environment variables for build version"
+    exit 1
+fi
+
+BUILD_VERS=$OMS_BUILDVERSION_MAJOR.$OMS_BUILDVERSION_MINOR.$OMS_BUILDVERSION_PATCH-$OMS_BUILDVERSION_BUILDNR
+
+# We assume that our build share is mounted at: /mnt/ostcdata. Verify that
+# our build directory is under that as expected ...
+
+BUILD_BASEDIR=/mnt/ostcdata/OSTCData/Builds/omsagent/develop
+
+if [ ! -d $BUILD_BASEDIR ]; then
+    echo "Not finding Build share mounted at $BUILD_BASEDIR ..."
+    exit 1
+fi
+
+BUILD_BASEDIR=${BUILD_BASEDIR}/${BUILD_VERS}
+
+if [ ! -d $BUILD_BASEDIR ]; then
+    echo "Not finding build output for version $BUILD_VERS at: $BUILD_BASEDIR"
+    exit 1
+fi
+
+zip -L >/dev/null 2>&1
+if [ $? != 0 ]; then
+    echo "zip must be installed"
+    exit 1
+fi
+
+unzip -Z >/dev/null 2>&1
+if [ $? != 0 ]; then
+    echo "unzip must be installed"
+    exit 1
+fi
+
+# Drop into build directory for remainder of operations
+
+cd omsagent/build
+
+# Set definitions and sign
+
+IntermediateDir="../intermediate"
+TargetDir="../target"
+DscModuleIntermediateDir="$IntermediateDir/merging_dsc_modules"
+DscModuleTargetDir="$TargetDir/dsc_signed"
+X64Dir="$BUILD_BASEDIR/Linux_ULINUX_1.0_x64_64_Release/dsc"
+X86Dir="$BUILD_BASEDIR/Linux_ULINUX_1.0_x86_32_Release/dsc"
+TestSigningDir=$1
+
+if [ "$TestSigningDir" = "" ]
+then
```

Comment:
Yeah, that's because three hands were in the pie. only two of which were engineers. There were a bunch of occurrences of this; I changed all to our typical "if test; then" construct used in the vast majority of other code.